docs/react/3.optimization/error-boundaries.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Error Boundaries
+sidebar_label: Error Boundaries
+---
+
+import Img from '../../../src/components/Img';
+
+In the past, JavaScript errors inside components used to corrupt React’s internal state and cause it to emit cryptic errors on next renders. These errors were always caused by an earlier error in the application code, but React did not provide a way to handle them gracefully in components, and could not recover from them.
+
+## Introducing Error Boundaries
+
+A JavaScript error in a part of the UI shouldn’t break the whole app. To solve this problem for React users, React 16 introduces a new concept of an “error boundary”.
```

Comment:
Adding  comparison diagram will be intuitional. After had read the whole article, I realized the difference between the component wrapped by Error Boundaries and normal component when the errors occurred inside.